playbooks/openstack-deploy_caasp.yml

```diff
@@ -183,18 +183,31 @@
 
         - meta: refresh_inventory
 
-        # Waiting for port is not enough, as the port might be up (systemd)
-        # while ssh is not ready. Wait for connection is not good either
-        # because the key is unknown.
-        - name: Wait for a while for SSH to be properly booted up
+        - name: Wait for a SSH to be properly started
           wait_for:
-            timeout: 120
+            host: "{{ hostvars[item]['ansible_host'] }}"
+            port: 22
+            search_regex: OpenSSH
+            state: started
+            delay: 10
+            timeout: 180
+          loop: "{{ groups['caasp-admin'] + groups['caasp-masters'] + groups['caasp-workers'] }}"
```

Comment:
should we use a set here so we dont have the same node twice?

Is that even possible, that caasp admin and caasp master are the same for example?

Reply:
It is not possible. They are different roles, which don't overlap, IIRC.